tools/koch/koch.nim

```diff
@@ -287,6 +293,24 @@ proc binArchive(target: BinArchiveTarget, args: string) =
 
   archive(binaryArgs & " " & args)
 
+proc addArchive(b: var Builder) =
+  let
+    niminst = b.addExecutable("niminst", "tools/niminst/niminst.nim")
+
+  let
+    defaultArgs = [
+      "--var:version=" & VersionAsString,
+      "--var:mingw=none",
+      "--main:" & (nimSource / "compiler" / "nim.nim")
+    ]
+
+  let compilerManifest = nimSource / "compiler" / "installer.ini"
+
+  let buildScript: Target = b.addRun("installscripts", niminst, @defaultArgs & @["scripts", compilerManifest])
+  discard b.addRun("archive", niminst, @defaultArgs & @["--format:tar.xz", "archive", compilerManifest], dependsOn = {buildScript})
```

Comment:
few cases where I think discardable makes sense, I don't think it catches errors here; because the dependency passing will force that.